x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/bulk_enable_disable_rules.ts

```diff
@@ -64,23 +64,17 @@ export const bulkEnableDisableRules = async ({
       ? await rulesClient.bulkEnableRules({ ids: ruleIds })
       : await rulesClient.bulkDisableRules({ ids: ruleIds });
 
-  const failedRuleIds = results.errors.map(({ rule: { id } }) => id);
-
-  // We need to go through the original rules array and update rules that were
-  // not returned as failed from the bulkEnableRules. We cannot rely on the
-  // results from the bulkEnableRules because the response is not consistent.
-  // Some rules might be missing in the response if they were skipped by
-  // Alerting Framework. See this issue for more details:
-  // https://github.com/elastic/kibana/issues/181050
   updatedRules.push(
```

Comment:
Correct me if I'm wrong by `updatedRules` could be removed completely. Since `results.rules` contains all rules now it should be used on line 96 like

```ts
return {
    updatedRules: results.rules,
    errors,
  };
```

The only this is that type casting is required `updatedRules: results.rules as RuleAlertType[]` since result has `BulkDisableRulesResult<Record<string, any>>` type which is default and comes from Kibana config schema in `x-pack/plugins/alerting/server/application/rule/schemas/rule_schemas.ts` but Security Solution uses stricter params type.

Reply:
yes, we kept `updatedRules ` so that we don't have to do type casting. But type casting is fine, will remove `updatedRules`

Reply:
[fixed](https://github.com/elastic/kibana/pull/191492/commits/da4ccc5c9851288042d66a94d8a3f8a22c4d1061#diff-eb24725306b8e25c36ac6f0303e6215117bfb19a8a0d8f7e3e003b9330e31ee2)